File: Find the closest pair from two arrays[27-September-2023]/Solution.java
//  >> JAVA CODE <<

class Solution{
    // Function for finding maximum and value pair
    public static ArrayList<Integer> printClosest (int arr[], int brr[], int n, int m, int x) {
        
        int val1=-1,val2=-1;
        int i=0,j=m-1;
        int diff=Integer.MAX_VALUE;
        
        while(i<n && j>=0)
        {
            int sum=arr[i]+brr[j];
            
            if(Math.abs(sum-x)<diff){
                diff=Math.abs(sum-x);
                val1=arr[i];
                val2=brr[j];
            }
            
            if(sum<=x) i++;
            else j--;
        }
        
        return new ArrayList(Arrays.asList(val1,val2));
        
        
    }
}

File: Find the closest pair from two arrays[27-September-2023]/solution.py
#   >> PYTHON CODE <<

class Solution:
    def printClosest (self,arr, brr, n, m, x) : 
        i,j=0,m-1
        diff=1e9
        val1,val2=-1,-1
        while(i<n and j>=0):
            sum=arr[i]+brr[j]
            
            if(abs(sum-x)<diff):
                diff=abs(sum-x)
                val1,val2=arr[i],brr[j]
                
            
            if(sum<=x):
                i+=1
            else:
                j-=1
        
        return [val1,val2]